scipy/stats/_sensitivity_analysis.py

```diff
@@ -0,0 +1,418 @@
+
+    :math:`S_{i}` corresponds to the first-order term which apprises the
+    contribution of the i-th parameter, while :math:`S_{ij}` corresponds to the
+    second-order term which informs about the correlations between the
+    i-th and the j-th parameters. These equations can be generalized to compute
+    higher order terms. However, the computational effort to converge them is
+    most often not at hand and their analysis and interpretations are not
+    simple. This is why only first order indices are provided.
+
+    Total indices represent the global contribution of the parameters on the
+    variance of the QoI and express as:
+
+    .. math::
+
+        S_{T_i} = S_i + \sum_j S_{ij} + \sum_{j,k} S_{ijk} + ...
+        = 1 - \frac{\mathbb{V}[\mathbb{E}(Y|x_{\sim i})]}{\mathbb{V}[Y]}.
+
+    The total number of function call is ``n+n+d*n=n(d+2)``.
+    Three matrices are required for the computation of
+    the indices: A, B and a permutation matrix AB based
+    on both A and B.
+
+    .. warning::
+
+        Negative Sobol' values are due to numerical errors. Increasing the
+        number of sample should help.
+
+    References
+    ----------
+    .. [1] Sobol, I. M.. "Sensitivity analysis for nonlinear mathematical
+       models." Mathematical Modeling and Computational Experiment, 1:407-414,
+       1993.
+    .. [2] Sobol, I. M. (2001). "Global sensitivity indices for nonlinear
+       mathematical models and their Monte Carlo estimates." Mathematics
+       and Computers in Simulation, 55(1-3):271-280,
+       :doi:`10.1016/S0378-4754(00)00270-6`, 2001.
+    .. [3] Saltelli, A. "Making best use of model evaluations to
+       compute sensitivity indices."  Computer Physics Communications,
+       145(2):280-297, :doi:`10.1016/S0010-4655(02)00280-1`, 2002.
+    .. [4] Saltelli, A., M. Ratto, T. Andres, F. Campolongo, J. Cariboni,
+       D. Gatelli, M. Saisana, and S. Tarantola. "Global Sensitivity Analysis.
+       The Primer." 2007.
+    .. [5] Saltelli, A., P. Annoni, I. Azzini, F. Campolongo, M. Ratto, and
+       S. Tarantola. "Variance based sensitivity analysis of model
+       output. Design and estimator for the total sensitivity index."
+       Computer Physics Communications, 181(2):259-270,
+       :doi:`10.1016/j.cpc.2009.09.018`, 2010.
+    .. [6] Ishigami, T. and T. Homma. "An importance quantification technique
+       in uncertainty analysis for computer models." IEEE,
+       :doi:`10.1109/ISUMA.1990.151285`, 1990.
+
+    Examples
+    --------
+    The following is an example with the Ishigami function [6]_
+
+    .. math::
+
+        Y(\mathbf{x}) = \sin x_1 + 7 \sin^2 x_2 + 0.1 x_3^4 \sin x_1,
+
+    with :math:`\mathbf{x} \in [-\pi, \pi]^3`. This function exhibits strong
+    non-linearity and non-monotonicity.
+
+    >>> import numpy as  np
+    >>> from scipy.stats import f_ishigami, sobol_indices, uniform
+    >>> rng = np.random.default_rng(1638083107694713882823079058616272161)
```

Comment:
```suggestion
    >>> rng = np.random.default_rng()
```
It's a pain putting this in and taking it out every time something changes, isn't it...

Reply:
Haha totally 😅